data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -171,18 +184,21 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
 
             final String groupsRaw = recordEvent.get(keyValueProcessorConfig.getSource(), String.class);
             final String[] groups = fieldDelimiterPattern.split(groupsRaw, 0);
+
+            parsedMap.putAll(defaultKeysMap);
```

Comment:
Will this logic conflict with skip duplicate values?

The overall solution here is:
1. Add the keys to the map so that they are there
2. Add keys you find later

This works for most cases. But, if there are conditions later which disallow adding the value, they would prevent this. One such condition is the allow duplicate values.

It seems that if you disallow duplicate values, this won't work as expected.

I think a safer approach would be to add the defaults at the end. It would require iterating over the values and checking that they are there or not. But, it would ensure that any conditions which disallow putting twice do not interfere with the behavior.